solutions/poi-18-stone.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: poi-18-stone
+source: POI 2018
+title: Pionek (stone)
+author: Albert Ye
+---
+
+## Solution
+This question can be solved through radial sweepline. Therefore, we sort our list of vectors by angle from the horizontal counterclockwise. This step comprises the bulk of the process, at $O(n \log n)$.
+
+We will now radially sweep from left to right. Note that from our current point, we only should search counterclockwise when adding vectors to prevent overcounting. If we searched both counterclockwise and clockwise, each orientation would be counted multiple times: Let us have two vectors $u$ and $v$ such that $u$ is before $v$ in the list. Then, if we searched counterclockwise, then searching from $u$ would detect $v$, and if we searched clockwise, then searching from $v$ would detect $u$. We do not want both of these events to occur, so we should only search in one direction. This turns the question into a sweepline problem, and we will be assuming that we are searching counterclockwise.
```

Comment:
```suggestion
We will now radially sweep from left to right. Note that from our current point, we only should search counterclockwise when adding vectors to prevent overcounting. If we searched both counterclockwise and clockwise, each orientation would be counted multiple times: Let us have two vectors $u$ and $v$ such that $u$ is before $v$ in the list. If we searched counterclockwise, then searching from $u$ would detect $v$, and if we searched clockwise, then searching from $v$ would detect $u$. We do not want both of these events to occur, so we should only search in one direction. This turns the question into a sweepline problem, and we will be assuming that we are searching counterclockwise.
```